### PR TITLE
Add explanation for implicit component attribute passing

### PR DESCRIPTION
**Pull request by chrisvfritz**
From https://github.com/vuejs/vue/issues/5846.

### CODE REVIEW

**Comment by chrisvfritz** [comment]
@posva Good call. 😄  I was looking [at a cat](https://en.wikipedia.org/wiki/Calico_cat) when making up that example. Updated!

File: src/v2/guide/components.md [hunk 1]
```diff
@@ -434,6 +434,44 @@ In addition, `type` can also be a custom constructor function and the assertion
 
 When a prop validation fails, Vue will produce a console warning (if using the development build).
 
+## Non-Prop Attributes
+
```
Comment (Austio) [inline src/v2/guide/components.md]: 👋 Hi, vue user here and followed to this pr from the issue discussion vuejs repo.

Have you considered adding a bottom line up front statement of the behavior and then the detail below that?  For instance, maybe a sentence here with something like.

```
A non-prop attribute is an attribute that is passed to a component that does not have a corresponding prop defined.

Non-prop attributes are passed as attributes to the root element of the component they are sent to.
```

Comment (Austio) [inline src/v2/guide/components.md]: btw, the documentation site is absolutely incredible, thank you for making it.

Comment (chrisvfritz) [inline src/v2/guide/components.md]: @Austio Great thinking. 🙂  I just updated the explanation.

**Pull request merged**